src/dune_rpc_impl/server.ml

```diff
@@ -321,23 +320,10 @@ let handler (t : _ t Fdecl.t) handle : 'a Dune_rpc_server.Handler.t =
   let () =
     let build _session targets =
       let server = Fdecl.get t in
-      match server.watch_mode_config with
-      | No -> assert false
-      | Yes Eager ->
-        let error =
-          Dune_rpc.Response.Error.create
-            ~kind:Invalid_request
-            ~message:
-              "the rpc server is running with eager watch mode using --watch. to run \
-               builds through an rpc client, start the server using --passive-watch-mode"
-            ()
-        in
-        raise (Dune_rpc.Response.Error.E error)
-      | Yes Passive ->
-        let ivar = Fiber.Ivar.create () in
-        let targets = List.map targets ~f:server.parse_build in
-        let* () = Job_queue.write server.pending_build_jobs (targets, ivar) in
-        Fiber.Ivar.read ivar
+      let ivar = Fiber.Ivar.create () in
+      let targets = List.map targets ~f:server.parse_build in
+      let* () = Job_queue.write server.pending_build_jobs (targets, ivar) in
+      Fiber.Ivar.read ivar
```

Comment:
To make sure I understand: you delete this part because either in `passive` or `eager` mode, we want to handle the request. Am I correct?

Reply:
That's correct. The only purpose of the `watch_mode_config` field was to prevent build requests from being handled by eager watch mode, so if we're allowing eager watch mode to handle build requests now then there's no need for the field or for this match statement (it now just always does what was formerly the `Yes Passive` case).